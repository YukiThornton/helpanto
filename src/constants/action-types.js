/* error-actions.js */
export const ENQUEUE_ERROR = 'ENQUEUE_ERROR';
export const DEQUEUE_ERROR = 'DEQUEUE_ERROR';

/* modal-actions.js */
export const OPEN_MODAL = 'OPEN_MODAL';
export const CLOSE_MODAL = 'CLOSE_MODAL';

/* recipe-actions.js */
export const SELECT_RECIPE_ON_LIST = 'SELECT_RECIPE_ON_LIST';
export const DESELECT_RECIPE_ON_LIST = 'DESELECT_RECIPE_ON_LIST';
export const FILTER_RECIPE_LIST = 'FILTER_RECIPE_LIST';
export const REQUEST_GET_RECIPES = 'REQUEST_GET_RECIPES';
export const REQUEST_GET_RECIPES_SUCCESS = 'REQUEST_GET_RECIPES_SUCCESS';
export const REQUEST_GET_RECIPES_FAILURE = 'REQUEST_GET_RECIPES_FAILURE';
export const REQUEST_POST_RECIPE = 'REQUEST_POST_RECIPE';
export const REQUEST_POST_RECIPE_SUCCESS = 'REQUEST_POST_RECIPE_SUCCESS';
export const REQUEST_POST_RECIPE_FAILURE = 'REQUEST_POST_RECIPE_FAILURE';
export const REQUEST_DELETE_RECIPE = 'REQUEST_DELETE_RECIPE';
export const REQUEST_DELETE_RECIPE_SUCCESS = 'REQUEST_DELETE_RECIPE_SUCCESS';
export const REQUEST_DELETE_RECIPE_FAILURE = 'REQUEST_DELETE_RECIPE_FAILURE';
