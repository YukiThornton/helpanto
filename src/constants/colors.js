export const BLUE_GRAY_100 = '#CFD8DC';
export const BLUE_GRAY_300 = '#90A4AE';
export const BLUE_GRAY_400 = '#78909C';
export const BLUE_GRAY_700 = '#455A64';
