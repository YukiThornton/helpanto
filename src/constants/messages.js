export const MSG_ERR_FETCH_RECIPES = 'Failed to get recipe lists. See you later.';
export const MSG_ERR_CREATE_RECIPE = 'Failed to create a new recipe. See you later.';
export const MSG_ERR_DELETE_RECIPE = 'Failed to delete a recipe. See you later.';
