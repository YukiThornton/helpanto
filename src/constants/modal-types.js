export const MODAL_TYPE_NEW_RECIPE = 'MODAL_TYPE_NEW_RECIPE';
export const MODAL_TYPE_NONE = 'MODAL_TYPE_NONE';
