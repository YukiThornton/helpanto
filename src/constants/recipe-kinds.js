export const MEMO = 'Memo';
export { MEMO as default };
