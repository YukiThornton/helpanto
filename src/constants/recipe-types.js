export const RECIPE_TYPE_MEMO = 'RECIPE_TYPE_MEMO';
export { RECIPE_TYPE_MEMO as default };
