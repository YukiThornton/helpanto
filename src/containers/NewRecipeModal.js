import React, { Component } from 'react';
import { connect } from 'react-redux';
import { createRecipe } from '../actions/recipe-actions';
import { closeModal, MODAL_TYPE_NEW_RECIPE } from '../actions/modal-actions';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import TextField from 'material-ui/TextField';
import FlatButton from 'material-ui/FlatButton';

class NewRecipeModal extends Component {
  static propTypes = {
    isOpen: PropTypes.bool.isRequired,
    onRequestClose: PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = {
      title: '',
      content: '',
    };
  }

  handleButtonClick = () => {
    this.props.onRequestCreateRecipe(this.state.title, this.state.content);
  };

  onClose = () => {
    this.setState({
      title: '',
      content: '',
    });
    this.props.onRequestClose();
  };

  render() {
    return (
      <Modal
        isOpen={this.props.isOpen}
        onRequestClose={this.onClose}
      >
        <h1>Create new recipe</h1>
        <MuiThemeProvider>
          <TextField
            hintText="Recipe title"
            value={this.state.title}
            onChange={(e, val) => this.setState({title: val})}
          />
        </MuiThemeProvider>
        <br />
        <br />
        <MuiThemeProvider>
          <TextField
            multiLine={true}
            rows={4}
            rowsMax={6}
            hintText="Recipe content"
            value={this.state.content}
            onChange={(e, val) => this.setState({content: val})}
          />
        </MuiThemeProvider>
        <MuiThemeProvider>
          <FlatButton
            onClick={this.handleButtonClick}
            label="Create"
            fullWidth={true}
          />
        </MuiThemeProvider>
        <MuiThemeProvider>
          <FlatButton
            onClick={this.onClose}
            label="Cancel"
            fullWidth={true}
          />
        </MuiThemeProvider>
      </Modal>
    );
  }
}

const isOpen = state => {
  return state.status.modal.isOpen && state.status.modal.modalType === MODAL_TYPE_NEW_RECIPE;
}

const mapStateToProps = state => {
  return {
    isOpen: isOpen(state),
  };
}

const mapDispatchToProps = dispatch => {
  return {
    onRequestClose: () => {
      dispatch(closeModal());
    },
    onRequestCreateRecipe: (title, content) => {
      dispatch(createRecipe(title, content));
      dispatch(closeModal());
    }
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NewRecipeModal);
