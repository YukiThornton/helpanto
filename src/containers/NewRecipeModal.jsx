import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import TextField from 'material-ui/TextField';
import FlatButton from 'material-ui/FlatButton';
import { createRecipe, fetchRecipes } from '../actions/recipe-actions';
import { closeModal } from '../actions/modal-actions';
import { MODAL_TYPE_NEW_RECIPE } from '../constants/modal-types';

export class NewRecipeModal extends Component {
  static propTypes = {
    isOpen: PropTypes.bool.isRequired,
    onRequestClose: PropTypes.func.isRequired,
    onRequestCreateRecipe: PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = {
      title: '',
      content: '',
    };
  }

  clearInput = () => {
    this.setState({
      title: '',
      content: '',
    });
  };

  handleSubmit = () => {
    this.props.onRequestCreateRecipe(this.state.title, this.state.content);
    this.clearInput();
  };

  handleRequestClose = () => {
    this.clearInput();
    this.props.onRequestClose();
  };

  handleTextInput = (input) => {
    this.setState(input);
  };

  render() {
    return (
      <Modal
        isOpen={this.props.isOpen}
        onRequestClose={this.handleRequestClose}
      >
        <h1>Create new recipe</h1>
        <TextField
          hintText="Recipe title"
          value={this.state.title}
          onChange={(e, val) => this.handleTextInput({ title: val })}
        />
        <br />
        <br />
        <TextField
          multiLine
          rows={4}
          rowsMax={6}
          hintText="Recipe content"
          value={this.state.content}
          onChange={(e, val) => this.handleTextInput({ content: val })}
        />
        <FlatButton
          onClick={this.handleSubmit}
          label="Create"
          fullWidth
        />
        <FlatButton
          onClick={this.handleRequestClose}
          label="Cancel"
          fullWidth
        />
      </Modal>
    );
  }
}

const isOpen = state =>
  state.status.modal.isOpen && state.status.modal.modalType === MODAL_TYPE_NEW_RECIPE;

export const mapStateToProps = (state) => {
  return {
    isOpen: isOpen(state),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    onRequestClose: () => {
      dispatch(closeModal());
    },
    onRequestCreateRecipe: (title, content) => {
      dispatch(createRecipe(title, content)).then(() => {
        dispatch(fetchRecipes());
      });
      dispatch(closeModal());
    },
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(NewRecipeModal);
