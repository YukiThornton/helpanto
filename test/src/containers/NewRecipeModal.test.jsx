import React from 'react';
import { shallow } from 'enzyme';
import { NewRecipeModal, mapStateToProps } from '../../../src/containers/NewRecipeModal';
import { MODAL_TYPE_NEW_RECIPE, MODAL_TYPE_NONE } from '../../../src/constants/modal-types';

describe('NewRecipeModal', () => {
  describe('mapStateToProps', () => {
    it('should return isOpen as true when MODAL_TYPE_NEW_RECIPE should open', () => {
      const store = {
        status: {
          modal: {
            isOpen: true,
            modalType: MODAL_TYPE_NEW_RECIPE,
          },
        },
      };
      expect(mapStateToProps(store)).toEqual({
        isOpen: true,
      });
    });

    it('should return isOpen as false when a modal (not MODAL_TYPE_NEW_RECIPE) should open', () => {
      const store = {
        status: {
          modal: {
            isOpen: true,
            modalType: MODAL_TYPE_NONE,
          },
        },
      };
      expect(mapStateToProps(store)).toEqual({
        isOpen: false,
      });
    });

    it('should return isOpen as false when no modal should open', () => {
      const store = {
        status: {
          modal: {
            isOpen: false,
            modalType: MODAL_TYPE_NONE,
          },
        },
      };
      expect(mapStateToProps(store)).toEqual({
        isOpen: false,
      });
    });
  });
  describe('NewRecipeModal (as Component)', () => {
    it('should be a visible modal', () => {
      const component = (
        <NewRecipeModal
          isOpen
          onRequestClose={() => {}}
          onRequestCreateRecipe={() => {}}
        />
      );
      expect(shallow(component)).toMatchSnapshot();
    });

    it('should be an invisible modal', () => {
      const component = (
        <NewRecipeModal
          isOpen={false}
          onRequestClose={() => {}}
          onRequestCreateRecipe={() => {}}
        />
      );
      expect(shallow(component)).toMatchSnapshot();
    });
  });
});
