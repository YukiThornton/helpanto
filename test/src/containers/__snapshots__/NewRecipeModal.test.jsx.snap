// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewRecipeModal NewRecipeModal (as Component) should be a visible modal 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h1>
    Create new recipe
  </h1>
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    fullWidth={false}
    hintText="Recipe title"
    multiLine={false}
    onChange={[Function]}
    rows={1}
    type="text"
    underlineShow={true}
    value=""
  />
  <br />
  <br />
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    fullWidth={false}
    hintText="Recipe content"
    multiLine={true}
    onChange={[Function]}
    rows={4}
    rowsMax={6}
    type="text"
    underlineShow={true}
    value=""
  />
  <FlatButton
    disabled={false}
    fullWidth={true}
    label="Create"
    labelPosition="after"
    labelStyle={Object {}}
    onClick={[Function]}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={false}
    secondary={false}
  />
  <FlatButton
    disabled={false}
    fullWidth={true}
    label="Cancel"
    labelPosition="after"
    labelStyle={Object {}}
    onClick={[Function]}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={false}
    secondary={false}
  />
</Modal>
`;

exports[`NewRecipeModal NewRecipeModal (as Component) should be an invisible modal 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h1>
    Create new recipe
  </h1>
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    fullWidth={false}
    hintText="Recipe title"
    multiLine={false}
    onChange={[Function]}
    rows={1}
    type="text"
    underlineShow={true}
    value=""
  />
  <br />
  <br />
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    fullWidth={false}
    hintText="Recipe content"
    multiLine={true}
    onChange={[Function]}
    rows={4}
    rowsMax={6}
    type="text"
    underlineShow={true}
    value=""
  />
  <FlatButton
    disabled={false}
    fullWidth={true}
    label="Create"
    labelPosition="after"
    labelStyle={Object {}}
    onClick={[Function]}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={false}
    secondary={false}
  />
  <FlatButton
    disabled={false}
    fullWidth={true}
    label="Cancel"
    labelPosition="after"
    labelStyle={Object {}}
    onClick={[Function]}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={false}
    secondary={false}
  />
</Modal>
`;
